docs/can-guides/upgrade/using-codemods.md

```diff
@@ -0,0 +1,122 @@
+@page guides/upgrade/using-codemods Using Codemods
+@parent guides/upgrade 1
+@description Learn how to migrate your app to CanJS 3 using [can-migrate-codemods](https://www.npmjs.com/package/can-migrate-codemods).
+
+@body
+
+## Overview
+
+A codemod is a transformation script that parses the [AST](https://en.wikipedia.org/wiki/Abstract_syntax_tree)
+of source code in order to do a code-aware find-and-replace refactor across
+multiple files. [can-migrate-codemods](https://www.npmjs.com/package/can-migrate-codemods)
+is a CLI utility for running codemods that can help migrate your app to CanJS 3.
+
+For example, the following CanJS 2.3 code:
+
+```js
+import can from "can";
+import "can/map/define/define";
+
+export default can.Map.extend({
+  define: {}
+});
+```
+
+…can be transformed to this:
+
+```js
+import CanMap from "can-map";
+import can from "can";
+import "can-map-define";
+
+export default CanMap.extend({
+  define: {}
+});
+```
+@highlight 1,3,5
+
+Using this CLI will get you about 85% of the way to having your codebase
+migrated; it’s not a complete solution for a seamless migration, but it will get
+you significantly closer than doing the migration by hand. Your own custom
+codemod transformations can also be used with this CLI.
+
+## Install
+
+Install `can-migrate-codemods` from npm:
+
+```shell
+npm install -g can-migrate-codemods
+```
+
+This will make the `can-migrate` command available globally.
+
+## Usage
+
+The CLI provides the following options:
+
+```
+can-migrate [<files> OPTIONS...]
+
+Updates files according to the CanJS 3 migration paths (minimal, modern, latest & greatest)
+
+Options
+--apply     -a    Apply transforms (instead of a dry run)
+--force           Apply transforms regardless of git status
+--silent    -s    Silence output
+--config    -c    Path to custom config file
+--transform -t    specify a transform
+```
+
+### Example
+
+Runs all the default `can-migrate` transforms on the files that match the `**/*.js` glob:
+
+```shell
+can-migrate --apply **/*.js
+```
+
+Runs the `can-component-rename` transform on the files that match the `**/*.js` glob:
+
+```shell
+can-migrate **/*.js --transforms can-component-rename/can-component-rename.js --apply
+```
+
+You can find a [complete list of transforms on GitHub](https://github.com/canjs/can-migrate-codemods/tree/master/src/transforms).
```

Comment:
Filed an issue to actually document the tranforms: https://github.com/canjs/can-migrate-codemods/issues/20